### ISSUE TITLE
Eval output during training generates aux as RGB

### ISSUE THREAD

**Issue by pabssen1** [wileewang/TransPixeler#27]
During training, the eval outputs are not as expected. Even after 1400 steps, the `_aux` output is being generated as RGB rather than alpha.

`dataset.toml`
`
 # Resolutions to train on, given as the side length of a square image. You can have multiple sizes here.
# !!!WARNING!!!: this might work differently to how you think it does. Images are first grouped to aspect ratio
# buckets, then each image is resized to ALL of the areas specified by the resolutions list. This is a way to do
# multi-resolution training, i.e. training on multiple total pixel areas at once. Your dataset is effectively duplicated
# as many times as the length of this list.
# If you just want to use predetermined (width, height, frames) size buckets, see the example cosmos_dataset.toml
# file for how you can do that.
# resolutions = [512]

# You can give resolutions as (width, height) pairs also. This doesn't do anything different, it's just
# another way of specifying the area(s) (i.e. total number of pixels) you want to train on.
resolutions = [[832, 480]]

# Enable aspect ratio bucketing. For the different AR buckets, the final size will be such that
# the areas match the resolutions you configured above.
enable_ar_bucket = true

# The aspect ratio and frame bucket settings may be specified for each [[directory]] entry as well.
# Directory-level settings will override top-level settings.

# Min and max aspect ratios, given as width/height ratio.
# min_ar = 0.5
# max_ar = 2.0
# Total number of aspect ratio buckets, evenly spaced (in log space) between min_ar and max_ar.
# num_ar_buckets = 7

# Can manually specify ar_buckets instead of using the range-style config above.
# Each entry can be width/height ratio, or (width, height) pair. But you can't mix them, because of TOML.
ar_buckets = [[832, 480]]
# ar_buckets = [1.0, 1.5]

# For video training, you need to configure frame buckets (similar to aspect ratio buckets). There will always
# be a frame bucket of 1 for images. Videos will be assigned to the first frame bucket that the video is greater than or equal to in length.
# But videos are never assigned to the image frame bucket (1); if the video is very short it would just be dropped.
frame_buckets = [105]
# If you have >24GB VRAM, or multiple GPUs and use pipeline parallelism, or lower the spatial resolution, you could maybe train with longer frame buckets
# frame_buckets = [1, 33, 65, 97]

[[directory]]
# Path to directory of images/videos, and corresponding caption files. The caption files should match the media file name, but with a .txt extension.
# A missing caption file will log a warning, but then just train using an empty caption.
mode = "pair"
path = '/workspace/TransPixeler/dataset'
# The dataset will act like it is duplicated this many times.
num_repeats = 1
# Example of overriding some settings, and using ar_buckets to directly specify ARs.
ar_buckets = [[832, 480]]
resolutions = [[832, 480]]
frame_buckets = [105]
shuffle=true

# You can list multiple directories.

# [[directory]]
# path = '/home/anon/data/images/something_else'
# num_repeats = 5

`

`train.toml`
`
# Output path for training runs. Each training run makes a new directory in here.
output_dir = 'video-generation-runs/rgba-wan-partial-lora'

# Dataset config file.
dataset = 'configs/dataset.toml'
# You can have separate eval datasets. Give them a name for Tensorboard metrics.
eval_datasets = [
    {name = 'something', config = 'configs/eval_dataset.toml'},
]

# training settings

# I usually set this to a really high value because I don't know how long I want to train.
epochs = 100
# Batch size of a single forward/backward pass for one GPU.
micro_batch_size_per_gpu = 2
# Pipeline parallelism degree. A single instance of the model is divided across this many GPUs.
pipeline_stages = 1
# Number of micro-batches sent through the pipeline for each training step.
# If pipeline_stages > 1, a higher GAS means better GPU utilization due to smaller pipeline bubbles (where GPUs aren't overlapping computation).
gradient_accumulation_steps = 1
# Grad norm clipping.
gradient_clipping = 1.0
# Learning rate warmup.
warmup_steps = 100

ema_decay = 0.9

# eval settings

eval_every_n_epochs = 100
eval_every_n_steps = 200
eval_before_first_step = false
# Might want to set these lower for eval so that less images get dropped (eval dataset size is usually much smaller than training set).
# Each size bucket of images/videos is rounded down to the nearest multiple of the global batch size, so higher global batch size means
# more dropped images. Usually doesn't matter for training but the eval set is much smaller so it can matter.
eval_micro_batch_size_per_gpu = 1
eval_gradient_accumulation_steps = 1

# misc settings

# Probably want to set this a bit higher if you have a smaller dataset so you don't end up with a million saved models.
save_every_n_epochs = 100
save_every_n_steps = 1000
# Can checkpoint the training state every n number of epochs or minutes. Set only one of these. You can resume from checkpoints using the --resume_from_checkpoint flag.
checkpoint_every_n_epochs = 100
# checkpoint_every_n_minutes = 120
# Always set to true unless you have a huge amount of VRAM.
activation_checkpointing = true
# Controls how Deepspeed decides how to divide layers across GPUs. Probably don't change this.
partition_method = 'parameters'
# dtype for saving the LoRA or model, if different from training dtype
save_dtype = 'bfloat16'
# Batch size for caching latents and text embeddings. Increasing can lead to higher GPU utilization during caching phase but uses more memory.
caching_batch_size = 1
# How often deepspeed logs to console.
steps_per_print = 1
# How to extract video clips for training from a single input video file.
# The video file is first assigned to one of the configured frame buckets, but then we must extract one or more clips of exactly the right
# number of frames for that bucket.
# single_beginning: one clip starting at the beginning of the video
# single_middle: one clip from the middle of the video (cutting off the start and end equally)
# multiple_overlapping: extract the minimum number of clips to cover the full range of the video. They might overlap some.
# default is single_middle
video_clip_mode = 'single_beginning'

[model]
type = 'wan-joint-partial-lora'
# Can load Hunyuan Video entirely from the ckpt path set up for the official inference scripts.
ckpt_path = '/workspace/TransPixeler/Wan2.1-T2V-1.3B'
# Or you can load it by pointing to all the ComfyUI files.
# transformer_path = '/data2/imagegen_models/hunyuan_video_comfyui/hunyuan_video_720_cfgdistill_fp8_e4m3fn.safetensors'
# vae_path = '/data2/imagegen_models/hunyuan_video_comfyui/hunyuan_video_vae_bf16.safetensors'
# llm_path = '/data2/imagegen_models/hunyuan_video_comfyui/llava-llama-3-8b-text-encoder-tokenizer'
# clip_path = '/data2/imagegen_models/hunyuan_video_comfyui/clip-vit-large-patch14'
# Base dtype used for all models.
dtype = 'bfloat16'
# Hunyuan Video supports fp8 for the transformer when training LoRA.
transformer_dtype = 'float8'
# How to sample timesteps to train on. Can be logit_normal or uniform.
timestep_sample_method = 'logit_normal'

[adapter]
type = 'lora'
rank = 256
alpha = 1
# Dtype for the LoRA weights you are training.
dtype = 'bfloat16'
# You can initialize the lora weights from a previously trained lora.
# init_from_existing = '/data/user/user69/projects/video-generation_runs/rgba/20250225_10-09-31/epoch1'
target_modules = [
    'self_attn',
    'cross_attn.q',
    'cross_attn.o',
    'ffn',
]

[optimizer]
# AdamW from the optimi library is a good default since it automatically uses Kahan summation when training bfloat16 weights.
# Look at train.py for other options. You could also easily edit the file and add your own.
type = 'adamw_optimi'
lr = 1e-3

betas = [0.9, 0.95]
weight_decay = 1e-4
eps = 1e-8
`

**Comment by wileewang** [comment]
The evaluation is performed on the EMA model, so you can observe the emergence of both modalities around 6,000–7,000 iterations.
Also, longer clips typically require more iterations to converge. We set frame_buckets = [33], and for the video dataset, each video has been split into multiple clips, resulting in a total of 3,000 video clips.

**Comment by pabssen1** [comment]
Thanks! So for custom datasets, as part of preprocessing, do we have to split the videos according to the frame_bucket we set?

**Comment by pabssen1** [comment]
Thanks! So should I set video_clip_mode = multiple_overlapping and frame_buckets = [33] and let it run for at least 6k steps? Also I am using my own dataset of rbg/alpha mp4. Are there any specific steps you would like to recommend for preprocessing?

**Comment by wileewang** [comment]
Yes, we recommend running for at least 6k steps for better results.

In our case, we processed the videos offline by splitting them into clips of 33 frames each and saving them directly. So we didn't set video_clip_mode in the training config.
